### PR TITLE
Added ElasticPagination class and used it in `quick_search()`. #6572

### PR DESCRIPTION
**Pull request by PritishC**

### CODE REVIEW

File: pagination.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+"""
+    pagination.py
+
+    :copyright: (c) 2014 by Openlabs Technologies & Consulting (P) Limited
```
Comment (sharoonthomas) [inline pagination.py]: its 2015 :-1:

File: pagination.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    :license: BSD, see LICENSE for more details.
+"""
+from nereid.contrib.pagination import Pagination
+from trytond.pool import Pool
+from werkzeug.utils import cached_property
+
+
+class ElasticPagination(Pagination):
```
Comment (sharoonthomas) [inline pagination.py]: Inherit from `BasePagination` please since you do not need the capabilities of ModelSQL pagination

File: pagination.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        """
+        :param model: Name of the tryton model on which the pagination is
+                      happening.
+        :param search_obj: The `~pyes.query.Search` object
+        :param page: The page number
+        :param per_page: Items per page
+        """
+        self.model_name = model
+        self.search_obj = search_obj
+        super(ElasticPagination, self).__init__(self.model, [], page, per_page)
```
Comment (sharoonthomas) [inline pagination.py]: will change since you have to inherit from `BasePagination`

File: pagination.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    @property
+    def configuration(self):
+        return Pool().get('elasticsearch.configuration')(1)
+
+    @cached_property
+    def count(self):
+        """
+        Returns count of items.
+        """
+        if self._count is not None:
```
Comment (sharoonthomas) [inline pagination.py]: who sets `_count` ???

File: pagination.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            doc_types=[self.configuration.make_type_name('product.product')]
+        ).count()
+
+    def items(self):
+        """
+        Returns items on the current page.
+        """
+        conn = self.configuration.get_es_connection(timeout=5)
+
+        result_set = conn.search(
```
Comment (sharoonthomas) [inline pagination.py]: self.result_set

File: pagination.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    _count = None
+
+    def __init__(self, model, search_obj, page, per_page):
+        """
+        :param model: Name of the tryton model on which the pagination is
+                      happening.
+        :param search_obj: The `~pyes.query.Search` object
+        :param page: The page number
+        :param per_page: Items per page
+        """
```
Comment (sharoonthomas) [inline pagination.py]: self.result_set = None

File: pagination.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+    def items(self):
+        """
+        Returns items on the current page.
+        """
+        conn = self.configuration.get_es_connection(timeout=5)
+
+        result_set = conn.search(
+            self.search_obj,
+            doc_types=[self.configuration.make_type_name('product.product')],
```
Comment (sharoonthomas) [inline pagination.py]: `s/'product.product'/self.model_name`

File: pagination.py (hunk 6 above)
Comment (PritishC) [inline pagination.py]: Now using it as a `@cached_property` as you suggested.

File: pagination.py [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        """
+        self.model_name = model
+        self.search_obj = search_obj
+        super(ElasticPagination, self).__init__(page, per_page)
+
+    @property
+    def model(self):
+        return Pool().get(self.model_name)
+
+    @property
```
Comment (sharoonthomas) [inline pagination.py]: remove

File: pagination.py [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+    @cached_property
+    def result_set(self):
+        """
+        Generates the `~pyes.es.ResultSet` object after performing the search.
+        """
+        conn = self.configuration.get_es_connection(timeout=5)
+
+        return conn.search(
+            self.search_obj,
```
Comment (sharoonthomas) [inline pagination.py]: add `start` and `size` based on `offset` and `per_page`

**Pull request merged**